tests/integration/test_studio_editable.py

```diff
@@ -0,0 +1,299 @@
+        comment_control.send_keys('forcing a change')
+        comment_control.clear()
+
+        self.click_save()
+        for field_name in EditableXBlock.editable_fields:
+            self.assertEqual(getattr(block, field_name), block.fields[field_name].default)
+            self.assertTrue(block.fields[field_name].is_set_on(block))
+
+    def test_set_and_reset(self):
+        """
+        Test that we can set values, save, then reset to defaults.
+        """
+        block = self.load_root_xblock()
+        for field_name in EditableXBlock.editable_fields:
+            self.assertFalse(block.fields[field_name].is_set_on(block))
+
+        for field_name in EditableXBlock.editable_fields:
+            color_control = self.get_element_for_field(field_name)
+            color_control.clear()
+            color_control.send_keys('1000')
+
+        self.click_save()
+
+        self.assertEqual(block.color, '1000')
+        self.assertEqual(block.count, 1000)
+        self.assertEqual(block.comment, '1000')
+
+        for field_name in EditableXBlock.editable_fields:
+            self.click_reset_for_field(field_name)
+
+        self.click_save()
+        block = self.load_root_xblock()  # Need to reload the block to bypass its cache
+        for field_name in EditableXBlock.editable_fields:
+            self.assertEqual(getattr(block, field_name), block.fields[field_name].default)
+            self.assertFalse(block.fields[field_name].is_set_on(block))
+
+    def test_invalid_data(self):
+        """
+        Test that we get notified when there's a problem with our data.
+        """
+        def assert_unchanged():
+            block = self.load_root_xblock()
+            for field_name in EditableXBlock.editable_fields:
+                self.assertEqual(getattr(block, field_name), block.fields[field_name].default)
+                self.assertFalse(block.fields[field_name].is_set_on(block))
+        def expect_error_message(expected_message):
+            notification = self.dequeue_runtime_notification()
+            self.assertEqual(notification[0], "error")
+            self.assertEqual(notification[1]["title"], "Unable to update settings")
+            self.assertEqual(notification[1]["message"], expected_message)
+
+        color_control = self.get_element_for_field('color')
+        color_control.clear()
+        color_control.send_keys('orange')
+
+        count_control = self.get_element_for_field('count')
+        count_control.clear()
+        count_control.send_keys('-10')
+
+        comment_control = self.get_element_for_field('comment')
+        comment_control.send_keys("That's a damn shame.")
+
+        self.click_save(expect_success=False)
+        expect_error_message("Count cannot be negative, No swearing allowed")
+        assert_unchanged()
+
+        count_control.clear()
+        count_control.send_keys('10')
+
+        self.click_save(expect_success=False)
+        expect_error_message("No swearing allowed")
```

Comment:
The count control is now good. But this error is on the `comment` control, which is still invalid as we haven't changed it since the last time we tried clicking "save".